apps/web/src/pages/templates/workflow/SideBar/SelectedStep.tsx

```diff
@@ -153,41 +131,31 @@ export function SelectedStep({
               </Text>
             </NavSection>
             <NavSection>
-              {steps.map((i, index) => {
-                return index === activeStep ? (
-                  <DigestMetadata
-                    key={i._id || i.id}
-                    control={control}
-                    index={index}
-                    loading={isLoading || isUpdateLoading}
-                    disableSubmit={readonly || loadingEditTemplate || isLoading || !isDirty}
-                    setSelectedChannel={setSelectedChannel}
-                  />
-                ) : null;
-              })}
+              <When truthy={hasActiveStepSelected}>
+                <DigestMetadata
+                  control={control}
+                  index={activeStepIndex}
+                  loading={isLoading || isUpdateLoading}
+                  disableSubmit={readonly || loadingEditTemplate || isLoading || !isDirty}
```

Comment:
Can we isolate this in a function or a variable?

Reply:
sure, good suggestion!